azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -156,7 +166,14 @@ def on_published():
                 logger.debug("{}({}): PUBACK received. completing op.".format(self.name, op.name))
                 op.complete()
 
-            self.transport.publish(topic=op.topic, payload=op.payload, callback=on_published)
+            try:
+                self.transport.publish(topic=op.topic, payload=op.payload, callback=on_published)
+            except (
+                transport_exceptions.ConnectionDroppedError,
+                transport_exceptions.ConnectionFailedError,
```

Comment:
>transport_exceptions.ConnectionFailedError, [](start = 16, length = 43)

According to documentation in `mqtt_transport.py`, this error is not one of the ones expected to be thrown. Is this a mistake here, or is there a mistake in `mqtt_transport.py`?

Reply:
I think you're right.  I'll go back and check.

---
In reply to: [388025540](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r388025540) [](ancestors = 388025540)